aptos-move/aptos-vm/src/move_vm_ext/session.rs

```diff
@@ -317,11 +340,11 @@ impl<'r, 'l> SessionExt<'r, 'l> {
 
         for (addr, account_changeset) in change_set.into_inner() {
             let (modules, resources) = account_changeset.into_inner();
-            for (struct_tag, blob_op) in resources {
+            for (struct_tag, blob_and_layout_op) in resources {
                 let state_key = StateKey::access_path(ap_cache.get_resource_path(addr, struct_tag));
                 let op = woc.convert_resource(
                     &state_key,
-                    blob_op,
+                    blob_and_layout_op,
```

Comment:
let's just do the following:
1. unpack `Op<(Bytes, Option<MoveTypeLayout>)>` into `(Op<Bytes>, Option<MoveTypeLayout>)`
2. We can then use the same flow to get a WriteOp, right? No need to make any changes, do clones, use Arcs.
3. The `op` from above will be a `WriteOp`, so then you pack it with the layout and add to resource write set.

Reply:
Is this change required anymore, as we are using Arc pointers anyways?